onnx/defs/object_detection/defs.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "onnx/defs/schema.h"
+using namespace ONNX_NAMESPACE;
+
+namespace ONNX_NAMESPACE {
+
+static const char* RoIAlign_ver1_doc = R"DOC(
+Region of Interest (RoI) align operation described in the
+[Mask R-CNN paper](https://arxiv.org/abs/1703.06870).
+RoIAlign consumes an input tensor X and region of interests (rois)
+to apply pooling across each RoI; it produces a 4-D tensor of shape
+(num_rois, C, output_height, output_width).
+
+RoIAlign is proposed to avoid the misalignment by removing
+quantizations while converting from original image into feature
+map and from feature map into RoI feature; in each ROI bin,
+the value of the sampled locations are computed directly
+through bilinear interpolation.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    RoIAlign,
+    10,
+    OpSchema()
+      .SetDoc(RoIAlign_ver1_doc)
+      .Attr(
+          "spatial_scale",
+          "Multiplicative spatial scale factor to translate ROI coordinates "
+          "from their input spatial scale to the scale used when pooling, "
+          "i.e., spatial scale of the input feature map X relative to the "
+          "input image. E.g.; default is 1.0f. ",
+          AttributeProto::FLOAT,
+          1.f)
+      .Attr(
+          "output_height",
+          "default 1; Pooled output Y's height.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "output_width",
+          "default 1; Pooled output Y's width.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "sampling_ratio",
+          "Number of sampling points in the interpolation grid used to compute "
+          "the output value of each pooled output bin. If > 0, then exactly "
+          "sampling_ratio x sampling_ratio grid points are used. If == 0, then "
+          "an adaptive number of grid points are used (computed as "
+          "ceil(roi_width / output_width), and likewise for height). Default is 0.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Input(
+          0,
+          "X",
+          "Input data tensor from the previous operator; "
+          "4-D feature map of shape (N x C x H x W), "
+          "where N is the batch size, C is the number of channels, "
+          "and H and W are the height and the width of the data.",
+          "T1")
+      .Input(
+          1,
+          "rois",
+          "RoIs (Regions of Interest) to pool over; rois is "
+          "2-D input of shape (num_rois, 4) given as "
+          "[[x1, y1, x2, y2], ...]. "
+          "The RoIs' coordinates are in the coordinate system of the input image.",
+          "T1")
+      .Input(
+          2,
+          "batch_splits",
+          "Tensor of shape (num_images) with each element denoting the number "
```

Comment:
This doesn't seem to support all cases in the current api design? Currently (both pytorch and onnx) rois are 2-D input of shape (num_rois, 5) given as [[batch_id, x1, y1, x2, y2], ...], which might not be sorted by batch_id. How about having the input as
```
.Input(
  2,
  "batch_indices",
  "Tensor of shape (num_rois) with each element denoting "
  "the index of the corresponding image in the batch.",
  "T2")
```

Reply:
Or we can just follow the pytorch spec ([[batch_id, x1, y1, x2, y2], ...]) for this? Will probably make conversions easier.

Reply:
@sampepose Can you update the documentation for this input? Also, since we'll be storing multiple indices, let's call it batch_indices like @BowenBao mentioned. I think this should be the final change.